docs/backend/widgets.md

```diff
@@ -472,6 +472,56 @@ self.widgets["ds_pregu_pers"].disabled = "disabled"
 ```
 
 
+### Customize existing widgets `pattern_options`
+
+There is a special attribute called `pattern_options` which is primarily used in Classic UI for pattern configuration.
+
+If you define your own schema, you can set this attribute with autoform directives.
+See {ref}`relations-configure-the-relateditemsfieldwidget-label` for an example.
+
+To customize or extend `pattern_options` for existing schema widgets, you can create a `z3c.form.interface.IValue` multiadapter as shown.
+
+```{code-block} python
+from z3c.form.interfaces import IForm
+from z3c.form.interfaces import IValue
+from z3c.form.interfaces import IWidget
+from zope.component impport adapter
+from zope.interface import implementer
+from zope.interface import Interface
+from zope.publisher.interfaces import  IRequest
+from zope.schema.interfaces import IField
+
+@implementer(IValue)
+@adapter(Interface, IRequest, IForm, IField, IWidget)
+class CustomPatternOptions:
+
+    def __init__(self, context, request, form, field, widget):
+        self.context = context
+        self.request = request
+        self.form = form
+        self.field = field
+        self.widget = widget
+
+    def get(self):
+        # return a dictionary with your custom options
+        return {
+            "myoption": "value",
+        }
+```
+
+Now register the adapter with ZCML as shown.
+
+```xml
+<adapter factory=".CustomPatternOptions"
+         name="pattern_options">
+```
+
+This registers the `pattern_options` for every field in every widget.
+
+You can define the adapter more explicitly by providing the fields or widgets interface in the `for` attribute.
```

Comment:
Where is the example of the adapter in the referenced section for the use of `for` attribute ?

The example show an adapter without `for` attribute:

```
<adapter factory=".ContentbrowserPatternOptions"
         name="pattern_options">
```

Reply:
Thanks for the review. We had discussions about the for attribute and therefore I moved it to the code. See https://github.com/plone/documentation/pull/1948#discussion_r2083875775 ... I'll change the text accordingly.